### ISSUE TITLE
[Mailer] messageId set after SentMessageEvent when using SmtpTransport 

### ISSUE THREAD

**Issue by tomme87** [symfony/symfony#52515]
### Symfony version(s) affected

6.3.7

### Description

When using the SMTP transport and listending for the `SentMessageEvent` the `SentMessage` contains the `messageId` generated by symfony instead of the `messageId` that is returned from the SMTP server.

This is because the code for setting the `messageId` is in the `send` method here:
https://github.com/symfony/symfony/blob/5f74215fed57df65a88427195589bdd020b4e8ec/src/Symfony/Component/Mailer/Transport/Smtp/SmtpTransport.php#L150-L152
witch is after the event is sent.

### How to reproduce

Send a email using the SMTP transport and listen for the `SentMessageEvent`. In this event, the `messageId` is wrong.

```php
<?php

use Symfony\Component\EventDispatcher\EventDispatcher;
use Symfony\Component\Mailer\Event\SentMessageEvent;
use Symfony\Component\Mailer\Transport;
use Symfony\Component\Mime\Email;

require_once 'vendor/autoload.php';

$dispatcher = new EventDispatcher();
$dispatcher->addListener(SentMessageEvent::class, function (SentMessageEvent $event) {
  var_dump($event->getMessage()->getMessageId());
});

$transport = Transport::fromDsn('smtps://usr:pw@email-smtp.eu-west-1.amazonaws.com', $dispatcher);

$email = (new Email())
  ->from('from@example.com')
  ->to('to@example.com')
  ->subject('Test')
  ->text('Testing');

$sentMessage = $transport->send($email);
var_dump($sentMessage->getMessageId()); // This is different from the messageId dumped in the listener!
```

### Possible Solution

Move the code for setting messageId into the `doSend` method?

**Comment by tomme87** [comment]
A workaround is to get `SentMessage` from the `WorkerMessageHandledEvent` instead (if using messenger).

```php
public function onWorkerMessageHandled(WorkerMessageHandledEvent $event): void {
  $sendEmailMessage = $event->getEnvelope()->getMessage();
  if (!$sendEmailMessage instanceof SendEmailMessage) {
    return;
  }

  $handledStamp = $event->getEnvelope()->last(HandledStamp::class);
  if (null === $handledStamp) {
    return;
  }

  $sentMessage = $handledStamp->getResult();
  if (!$sentMessage instanceof SentMessage) {
    return;
  }

  $messageId = $sentMessage->getMessageId();
}
```

**Comment by xabbuh** [comment]
Can you confirm that your issue would be fixed with #53969?

**Comment by tomme87** [comment]
> Can you confirm that your issue would be fixed with #53969?

Yes, that will fix my issue :+1:
